frame/system/src/lib.rs

```diff
@@ -951,12 +949,27 @@ impl<T: Trait> Module<T> {
 		<ExtrinsicsRoot<T>>::put(txs_root);
 
 		if let InitKind::Full = kind {
+			sp_io::storage::accumulator_commit(&Events::<T>::storage_value_final_key()[..]);
 			<Events<T>>::kill();
 			EventCount::kill();
 			<EventTopics<T>>::remove_all();
 		}
 	}
 
+	/// Update pending storage changes, if any.
+	pub fn update_accumulators() {
+		sp_io::storage::accumulator_commit(&Events::<T>::storage_value_final_key()[..]);
+	}
+
+	/// Collect events for the current block.
+	///
+	/// This should not be used for anything but tests. Using it for anything else
```

Comment:
If it should only be used for tests let's just put a feature flag?

Reply:
We don't use feature flags for testing stuff. Maybe latter then features wouldn't leak in cargo.